### PR TITLE
Add nasa images

### LINKED ISSUE #233: Implement ISS API get function
As a user interested in tracking the International Space Station (ISS), I want to see the current location of the ISS in real-time within the app, so that I can stay informed about its position and know when it will pass over my location.

- Use `requests` library to get information on ISS

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #233 

- Written tests for ISS api function
- Implemented ISS api function

### CODE REVIEW

File: dashboard/test_image_pipeline.py [hunk 1]
```diff
@@ -124,11 +127,57 @@ def test_nasa_pipeline_no_image_found(self, mock_has_nasa_image, mock_get_nasa_i
...
+        '''Tests that the iss request function returns data in the correct format'''
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {
+            'message': 'success',
+            'timestamp': 1728908772,
+            'iss_position': {'latitude': '46.9566','longitude': '-3.7535'}}
+        mock_get.return_value = mock_response
+        result = get_iss_location()
+        expected_result = {'timestamp': dt(2024,10,14,13,26,12),
```
Comment (AlexOBarnes) [inline dashboard/test_image_pipeline.py:154]:
```suggestion
        expected_result = {'timestamp': dt(2024,10,14,12,26,12),
```

File: dashboard/image_pipeline.py [hunk 2]
```diff
@@ -68,25 +70,26 @@ def nasa_pipeline() -> None:
...
-def iss_pipeline() -> None:
-    '''Runs the ISS pipeline to obtain the current ISS location'''
-    ...
+    raise APIError('Unsuccessful request.', response.status_code)
 
 if __name__ == '__main__':
     logger = logging.getLogger(__name__)
     logging.basicConfig(level=logging.INFO)
     load_dotenv()
-    nasa_pipeline()
```
Comment (olliejth) [inline dashboard/image_pipeline.py:92]: Nice work keeping the scripts clean.

File: dashboard/image_pipeline.py [hunk 3]
```diff
@@ -68,25 +70,26 @@ def nasa_pipeline() -> None:
...
+
 def get_iss_location() -> dict:
     '''Requests the current ISS location'''
-    ...
-
-def transform_iss_data() -> list:
-    '''Formats the ISS data ready for insertion'''
-    ...
+    url = 'http://api.open-notify.org/iss-now.json'
+    response = get(url, timeout=10)
```
Comment (olliejth) [inline dashboard/image_pipeline.py:80]: Very elegant.

File: dashboard/image_pipeline.py [hunk 4]
```diff
@@ -68,25 +70,26 @@ def nasa_pipeline() -> None:
...
-def load_iss_data() -> None:
-    '''Uploads the data into the database'''
-    ...
+    if response.status_code == 200:
+        data = response.json()
 
+        if data['message'] == 'success':
+            return {'timestamp':extract_unix_time(data['timestamp']),
+                    'latitude': data['iss_position']['latitude'],
+                    'longitude':data['iss_position']['longitude']}
```
Comment (olliejth) [inline dashboard/image_pipeline.py:89]: Clean and efficient.

File: dashboard/test_image_pipeline.py [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
-import pytest
+#pylint: disable=C0121,R0903
 from unittest.mock import patch, MagicMock
 from datetime import datetime as dt
+import pytest
 from image_pipeline import (has_nasa_image,extract_time,get_nasa_image,
-    load_image,nasa_pipeline,APIError)
+    load_image,nasa_pipeline,APIError, get_iss_location)
 
 NASA_API_URL = 'https://api.nasa.gov/planetary/apod?api_key=YOUR_API_KEY'
```
Comment (olliejth) [inline dashboard/test_image_pipeline.py:9]: Do you need to remove "YOUR_API_KEY" from this?

Comment (olliejth) [reply dashboard/test_image_pipeline.py:9]: Is this used anywhere in the test file?

**Review by olliejth — CHANGES_REQUESTED** [review]
Left some comments on dashboard/test_image_pipeline.py. May not be an issue but worth a check.

**Comment by AlexOBarnes** [comment]
Yeah, I see what your saying, it is used to mock the api call in some of the previous tests that weren't changed in this PR

File: dashboard/image_pipeline.py [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
-sys.path.append(os.path.abspath(os.path.join(
-    os.path.dirname(__file__), '..', 'weekly-openmeteo')))
-from os import environ as ENV
 import logging
 from datetime import datetime as dt
 from dotenv import load_dotenv
 from requests import get
-from psycopg2 import connect
+
+sys.path.append(os.path.abspath(os.path.join(
```
Comment (MurasatoNaoya) [inline dashboard/image_pipeline.py:11]: A good way of reducing redundancy, not having to copy an existing module.

File: dashboard/test_image_pipeline.py [hunk 7]
```diff
@@ -124,11 +127,57 @@ def test_nasa_pipeline_no_image_found(self, mock_has_nasa_image, mock_get_nasa_i
...
 
         nasa_pipeline()
 
         mock_get_nasa_image.assert_not_called()
         mock_load_image.assert_not_called()
+
+
+class TestGetIssLocation:
+    '''Contains tests for get iss location'''
+    @patch('image_pipeline.get')
```
Comment (MurasatoNaoya) [inline dashboard/test_image_pipeline.py:143]: Good use of patching to to interact with the actual ISS API.

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.

**Review by olliejth — APPROVED** [review]
Changes explained and resolved.

**Pull request merged**